# Melbourne Housing Market

The classic Data Science challenge: Analysis and prediction of housing prices in Melbourne, Australia  
source: [Kaggle](https://www.kaggle.com/datasets/anthonypino/melbourne-housing-market)

## 0. Imports

In [14]:
import pandas as pd

In [15]:
# Set pandas options
pd.set_option('display.float_format', '{:.2f}'.format)

## 1. Data analysis

In [22]:
data = pd.read_csv('Melbourne_housing_FULL.csv')
data_raw = data.copy()

In [17]:
data_raw.shape

(34857, 21)

In [18]:
data_raw.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 34857 entries, 0 to 34856
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Suburb         34857 non-null  object 
 1   Address        34857 non-null  object 
 2   Rooms          34857 non-null  int64  
 3   Type           34857 non-null  object 
 4   Price          27247 non-null  float64
 5   Method         34857 non-null  object 
 6   SellerG        34857 non-null  object 
 7   Date           34857 non-null  object 
 8   Distance       34856 non-null  float64
 9   Postcode       34856 non-null  float64
 10  Bedroom2       26640 non-null  float64
 11  Bathroom       26631 non-null  float64
 12  Car            26129 non-null  float64
 13  Landsize       23047 non-null  float64
 14  BuildingArea   13742 non-null  float64
 15  YearBuilt      15551 non-null  float64
 16  CouncilArea    34854 non-null  object 
 17  Lattitude      26881 non-null  float64
 18  Longti

In [19]:
data_raw.head()

Suburb             Address  Rooms Type      Price Method SellerG  \
0  Abbotsford       68 Studley St      2    h        NaN     SS  Jellis   
1  Abbotsford        85 Turner St      2    h 1480000.00      S  Biggin   
2  Abbotsford     25 Bloomburg St      2    h 1035000.00      S  Biggin   
3  Abbotsford  18/659 Victoria St      3    u        NaN     VB  Rounds   
4  Abbotsford        5 Charles St      3    h 1465000.00     SP  Biggin   

        Date  Distance  Postcode  ...  Bathroom  Car  Landsize  BuildingArea  \
0  3/09/2016      2.50   3067.00  ...      1.00 1.00    126.00           NaN   
1  3/12/2016      2.50   3067.00  ...      1.00 1.00    202.00           NaN   
2  4/02/2016      2.50   3067.00  ...      1.00 0.00    156.00         79.00   
3  4/02/2016      2.50   3067.00  ...      2.00 1.00      0.00           NaN   
4  4/03/2017      2.50   3067.00  ...      2.00 0.00    134.00        150.00   

   YearBuilt         CouncilArea Lattitude  Longtitude             Regionname  \
0        NaN  Yarra City Council    -37.80      145.00  Northern Metropolitan   
1        NaN  Yarra City Council    -37.80      145.00  Northern Metropolitan   
2    1900.00  Yarra City Council    -37.81      144.99  Northern Metropolitan   
3        NaN  Yarra City Council    -37.81      145.01  Northern Metropolitan   
4    1900.00  Yarra City Council    -37.81      144.99  Northern Metropolitan   

  Propertycount  
0       4019.00  
1       4019.00  
2       4019.00  
3       4019.00  
4       4019.00  

[5 rows x 21 columns]

In [21]:
data_raw.isnull().sum()

Suburb               0
Address              0
Rooms                0
Type                 0
Price             7610
Method               0
SellerG              0
Date                 0
Distance             1
Postcode             1
Bedroom2          8217
Bathroom          8226
Car               8728
Landsize         11810
BuildingArea     21115
YearBuilt        19306
CouncilArea          3
Lattitude         7976
Longtitude        7976
Regionname           3
Propertycount        3
dtype: int64

In [28]:
# Splitting of data into numerical and categorical data
data_raw_num = data_raw.select_dtypes(include=['int64', 'float64'])
data_raw_cat = data_raw.select_dtypes(include=['object'])
print(f"shape of data_raw_num:  {data_raw_num.shape}")
print(f"shape of data_raw_cat:  {data_raw_cat.shape}")
print(f"shape of full dataset : {data_raw.shape}")

shape of data_raw_num:  (34857, 13)
shape of data_raw_cat:  (34857, 8)
shape of full dataset : (34857, 21)


In [29]:
data_raw_num.describe()

Rooms       Price  Distance  Postcode  Bedroom2  Bathroom      Car  \
count 34857.00    27247.00  34856.00  34856.00  26640.00  26631.00 26129.00   
mean      3.03  1050173.34     11.18   3116.06      3.08      1.62     1.73   
std       0.97   641467.13      6.79    109.02      0.98      0.72     1.01   
min       1.00    85000.00      0.00   3000.00      0.00      0.00     0.00   
25%       2.00   635000.00      6.40   3051.00      2.00      1.00     1.00   
50%       3.00   870000.00     10.30   3103.00      3.00      2.00     2.00   
75%       4.00  1295000.00     14.00   3156.00      4.00      2.00     2.00   
max      16.00 11200000.00     48.10   3978.00     30.00     12.00    26.00   

       Landsize  BuildingArea  YearBuilt  Lattitude  Longtitude  Propertycount  
count  23047.00      13742.00   15551.00   26881.00    26881.00       34854.00  
mean     593.60        160.26    1965.29     -37.81      145.00        7572.89  
std     3398.84        401.27      37.33       0.09        0.12        4428.09  
min        0.00          0.00    1196.00     -38.19      144.42          83.00  
25%      224.00        102.00    1940.00     -37.86      144.93        4385.00  
50%      521.00        136.00    1970.00     -37.81      145.01        6763.00  
75%      670.00        188.00    2000.00     -37.75      145.07       10412.00  
max   433014.00      44515.00    2106.00     -37.39      145.53       21650.00

Some observations that will need further investigation:
* The minimum of 'Bedroom2' is 0.
* The minimum of 'Bathroom' is 0.
* The minimum of 'Landsize' is 0 [meters].
* The maximum of 'Landsize' is 4330141 [meters].
* The minimum value of 'BuildingArea' is 0.
* The maximum value of 'BuildingArea' is 44515 [meters].
* The minimum value of 'YearBuilt' is 1196. (Melbourne was founded in 1835)
* The maximum value of 'YearBuilt' is 2106. (Dataset was created in 2018)

In [46]:
print(f"Number of entries with bedroom2 < 1: {(data_raw_num['Bedroom2'] < 1).sum()}")
print(f"Number of entries with bathroom < 1: {(data_raw_num['Bathroom'] < 1).sum()}")
print(f"Number of entries with landsize = 0: {(data_raw_num['Landsize'] == 0).sum()}")
print(f"Number of entries with buildingarea = 0: {(data_raw_num['BuildingArea'] == 0).sum()}")
print(f"Number of entries with YearBuilt < 1835: {(data_raw_num['YearBuilt'] < 1835).sum()}")
print(f"Number of entries with YearBuilt > 2018: {(data_raw_num['YearBuilt'] > 2018).sum()}")
print(f"Number of entries with buildingarea > landsize: {(data_raw_num['BuildingArea'] > data_raw_num['Landsize']).sum()}")

Number of entries with bedroom2 < 1: 17
Number of entries with bathroom < 1: 46
Number of entries with landsize = 0: 2437
Number of entries with buildingarea = 0: 76
Number of entries with YearBuilt < 1835: 4
Number of entries with YearBuilt > 2018: 2
Number of entries with buildingarea > landsize: 1751


In [45]:
print(f"Number of entries with buildingarea = 0 and landsize = 0: {((data_raw_num['Landsize'] == 0) & (data_raw_num['BuildingArea'] == 0)).sum()}")
print(f"Number of entries with buildingarea > landsize and landsize = 0: {((data_raw_num['BuildingArea'] > data_raw_num['Landsize']) & (data_raw_num['Landsize'] == 0)).sum()}")

Number of entries with buildingarea = 0 and landsize = 0: 0
Number of entries with buildingarea > landsize and landsize = 0: 1327
